python/beeai_framework/workflows/agent.py

```diff
@@ -69,8 +69,24 @@ def del_agent(self, name: str) -> "AgentWorkflow":
 
     def add_agent(
         self,
-        agent: (BaseAgent | Callable[[ReadOnlyMemory], BaseAgent | asyncio.Future[BaseAgent]] | AgentFactoryInput),
+        agent: (
+            BaseAgent | Callable[[ReadOnlyMemory], BaseAgent | asyncio.Future[BaseAgent]] | AgentFactoryInput | None
+        ) = None,
+        /,
+        **kwargs: Any,
     ) -> "AgentWorkflow":
+        if agent is None or isinstance(agent, dict):
+            agent_args = kwargs.get("agent", agent or kwargs)
+            agent = (
+                (
+                    AgentFactoryInput.model_validate(agent_args, strict=False, from_attributes=True)
+                    if "name" in agent_args
```

Comment:
not sure if this is the best way to distinguish between arguments for `AgentFactoryInput` or `BaseAgent`. thoughts are welcomed